lib/core/src/Cardano/Wallet/Api.hs

```diff
@@ -304,6 +307,13 @@ type SelectCoins n = "wallets"
     :> ReqBody '[JSON] (ApiSelectCoinsDataT n)
     :> Post '[JSON] (ApiCoinSelectionT n)
 
+type SelectCoinsAction n = "wallets"
+    :> Capture "walletId" (ApiT WalletId)
+    :> "coin-selections"
+    :> "random"
+    :> ReqBody '[JSON] (ApiSelectCoinsActionDataT n)
+    :> Post '[JSON] (ApiCoinSelectionCertT n)
```

Comment:
Interesting. Does `servant` treats `:<|>` as a type-level Alternative now? I am pretty sure in the past that only the right-hand side would be picked by servant when used like this (with identical "signatures" on endpoints). If that acts as a sort-of type-level alternative then that's quite amazing and there are a few other API endpoints we can simplify :thinking:

Reply:
https://hackage.haskell.org/package/servant-0.18/docs/Servant-API.html#t::-60--124--62-